### PR TITLE
Python client docs, main structure refactor

### PR DESCRIPTION
**Pull request by maciej-flexcompute**
- Added python client docs. This is not meant to be written manually here but automatically parsed from docstrings from Flow360PythonClient module.
- Restructured main docs:
  - added: preprocessing, solver configuration, postprocessing

### CODE REVIEW

File: docs/source/capabilities/automatedMeshing.rst [hunk 1]
```diff
@@ -0,0 +1,264 @@
+.. _automatedMeshing:
+
+Automated Meshing
```
Comment (cjdoolittle) [inline docs/source/capabilities/automatedMeshing.rst:3]: recommend keeping automatedMeshing overview in Capabilities section and relocating/linking details to Preprocessing section

Comment (maciej-flexcompute) [reply docs/source/capabilities/automatedMeshing.rst]: moved

File: docs/source/preprocessing/automatedMeshing.rst [hunk 2]
```diff
@@ -0,0 +1,264 @@
+.. _automatedMeshing:
+
+Automated Meshing
+*************************
+
+Overview
+========
+
+Flow360 offers automated meshing, from CAD geometry through a surface mesh to volume mesh. The supported CAD format is \*.csm file which is Engineering Sketch Pad (EPS) format. The generated volume format is CGNS.
```
Comment (jbungenerflex) [inline docs/source/preprocessing/automatedMeshing.rst:9]: I would reword to: ...from CAD geometry to a surface mesh and finally to a volume mesh

File: docs/source/preprocessing/automatedMeshing.rst [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+Geometry
+========
+
+The Engineering Sketch Pad is a solid-modeling, feature-based, web-enabled system for building parametric geometry. It can be downloaded from `ESP's website <https://acdl.mit.edu/ESP/>`_.
+The geometry in ESP is described in text \*.csm file containing all CAD instructions. See csm example\:
```
Comment (jbungenerflex) [inline docs/source/preprocessing/automatedMeshing.rst:15]: ..... described in A text ..... 
add the "a"

File: docs/source/preprocessing/automatedMeshing.rst [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+   Engineering Sketch Pad view of the geometry.
+
+
+.. _ESPtoSurfaceMeshSection:
+
+Surface Meshing
+===============
+
+The surface mesher takes geometry file and configuration JSON file as input parameters and generates surface mesh. The meshing configuration file (or dict in python), also called surfaceMesh.json, contains information such as maximum element edge length, curvature resolution angle or growth rate of 2D layers. See JSON example:
```
Comment (jbungenerflex) [inline docs/source/preprocessing/automatedMeshing.rst:41]: ...takes THE geometry....
.... generates A surface mesh ....

File: docs/source/preprocessing/automatedMeshing.rst [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+- :code:`surfaceMeshId`
+
+For a full description of config.json for surface mesher see :ref:`here.<JSON surface mesher>`
+
+.. _SurfaceToVolumeMeshSection:
+
+Volume Meshing
+==============
+
+The volume mesher takes :code:`surfaceMeshId` and config JSON as arguments and generates CGNS mesh suitable for Flow360 solver. The JSON configuration file (or dict in python) specifies first layer thickness, growth rate, sizes and location of refinement zones (sources) and actuator disks. All geometry from ESP is treated as a no-slip wall therefore prism layers will be grown of geometry surfaces.
```
Comment (jbungenerflex) [inline docs/source/preprocessing/automatedMeshing.rst:79]: ...takes A surfacemeshId......
...generates A CGNS ....
...for THE Flow360.....
....specifies THE first layer .....
.... grown OFF of geometry surfaces....

File: docs/source/preprocessing/automatedMeshing.rst [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+For a full description of config.json for surface mesher see :ref:`here.<JSON surface mesher>`
+
+.. _SurfaceToVolumeMeshSection:
+
+Volume Meshing
+==============
+
+The volume mesher takes :code:`surfaceMeshId` and config JSON as arguments and generates CGNS mesh suitable for Flow360 solver. The JSON configuration file (or dict in python) specifies first layer thickness, growth rate, sizes and location of refinement zones (sources) and actuator disks. All geometry from ESP is treated as a no-slip wall therefore prism layers will be grown of geometry surfaces. 
+
+Farfield will be created automatically\:
```
Comment (jbungenerflex) [inline docs/source/preprocessing/automatedMeshing.rst:81]: THE farfield....

File: docs/source/preprocessing/volumeMesher.csv [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Option@ Type@ Default@ Example@ Description
+refinementFactor@ float@ 1@ 2@ If refinementFactor=r is provided all spacings will be adjusted to generate r-times finer mesh everywhere. For example, if refinementFactor=2, all spacings will be divided by :math:`2^{1/3}\approx 1.26` , so the resulting mesh will have approximately 2 times more cells.
```
Comment (jbungenerflex) [inline docs/source/preprocessing/volumeMesher.csv:2]: 2 times more cells or 2 times more nodes? We think in nodes here. right?

File: docs/source/preprocessing/volumeMesherSources.csv [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Option@ Type@ Default@ Example@ Description
+type@ [“box”, “cylinder”]@ “box”@ “box”@ Type of source. Supported box and cylinder. Cylinder’s axis is along the z-direction.
```
Comment (jbungenerflex) [inline docs/source/preprocessing/volumeMesherSources.csv:2]: instead of "Supported box and cylinder" say. Supported source shapes: Box and cylinder

File: docs/source/pythonAPI/flow360clientDoc.rst [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+   :param priority: Queueing priority, 'high' or 'low', by default 'high'
+   :type priority: str, optional
+   :param parentId: Parent case Id for forked cases. Format xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, by default None
+   :type parentId: str, optional
+   
+   :returns: Case id of format xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.
+   :rtype: str
+   
+
+.. py:function:: NewCaseListWithPhase(meshId, config, caseName=None, tags=[], priority='high', parentId=None, phaseCount=1)
```
Comment (jbungenerflex) [inline docs/source/pythonAPI/flow360clientDoc.rst:30]: We need to provide an example/description of when you would want to use this. It is not clear ot me why we wouldn't just launch 1 big case instead of many smaller dependent ones.

File: docs/source/pythonAPI/flow360clientDoc.rst [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+   
+   :param include_deleted: Whether to include deleted meshes in the list, by default False
+   :type include_deleted: bool, optional
+   
+   :returns: List of surface meshes
+   :rtype: list
+   
+
+.. py:module:: flow360client.task
+
```
Comment (jbungenerflex) [inline docs/source/pythonAPI/flow360clientDoc.rst:244]: What about the result download functions
DownloadResultsFile ( explain what the possible files to download are and what each of those file contains)
DownloadSurfaceResults
DownloadVolumetricResults

File: docs/source/pythonAPI/flow360clientDoc.rst (hunk 9 above)
Comment (maciej-flexcompute) [reply docs/source/pythonAPI/flow360clientDoc.rst:30]: I don't know either. I just described what I understood from the code. Any ideas?

File: docs/source/preprocessing/automatedMeshing.rst (hunk 2 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/automatedMeshing.rst:9]: done

File: docs/source/preprocessing/automatedMeshing.rst (hunk 3 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/automatedMeshing.rst:15]: done

File: docs/source/preprocessing/automatedMeshing.rst (hunk 4 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/automatedMeshing.rst:41]: done

File: docs/source/preprocessing/automatedMeshing.rst (hunk 5 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/automatedMeshing.rst:79]: done

File: docs/source/preprocessing/automatedMeshing.rst (hunk 6 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/automatedMeshing.rst:81]: done

File: docs/source/preprocessing/volumeMesherSources.csv (hunk 8 above)
Comment (maciej-flexcompute) [reply docs/source/preprocessing/volumeMesherSources.csv:2]: done

File: docs/source/pythonAPI/flow360clientDoc.rst (hunk 9 above)
Comment (jbungenerflex) [reply docs/source/pythonAPI/flow360clientDoc.rst:30]: @feilin-flexcompute emailed with a client that had the same question. HE would know.

Comment (maciej-flexcompute) [reply docs/source/pythonAPI/flow360clientDoc.rst:30]: improved

File: docs/source/pythonAPI/flow360clientDoc.rst (hunk 10 above)
Comment (maciej-flexcompute) [reply docs/source/pythonAPI/flow360clientDoc.rst]: added following functions:
GetCaseResidual()
GetCaseTotalForces()
GetCaseLinearResidual()
DownloadResultsFile()
DownloadVolumetricResults()
DownloadSurfaceResults()

File: docs/source/pythonAPI/flow360clientDoc.rst [hunk 11]
```diff
@@ -0,0 +1,380 @@
+flow360client module
+********************
+
+
+.. py:module:: flow360client
+
+
```
Comment (cjdoolittle) [inline docs/source/pythonAPI/flow360clientDoc.rst:7]: if feasible, can we add a tag for cross-referencing/linking from elsewhere in documents? For example, here:
`.. _NewCase:`

and, also for all other flow360client functions detailed.

**Review by feilin-flexcompute — APPROVED** [review]
